3D/Web/SRC/main.js

```diff
@@ -0,0 +1,279 @@
+import * as THREE from "three";
+import { OrbitControls } from "three/examples/jsm/controls/OrbitControls";
+//import object loader - this is for importing 3D models
+import { GLTFLoader } from "three/addons/loaders/GLTFLoader.js";
+//import textures
+import brown from "../Assets/Images/Textures/brown.jpg";
+import blue from "../Assets/Images/Textures/blue.jpg";
+import lightblue from "../Assets/Images/Textures/lightblue.jpg";
+import yellow from "../Assets/Images/Textures/yellow.jpg";
+import lightyellow from "../Assets/Images/Textures/lightyellow.jpg";
+
+//set up the scene
+const scene = new THREE.Scene();
+
+//set up the camera
+const camera = new THREE.PerspectiveCamera(
+    75,
+    window.innerWidth / window.innerHeight,
+    0.1,
+    1000
+);
+//we can tweak this position
+camera.position.setZ(25);
+camera.layers.enable(1);
+camera.layers.enable(2);
+
+//set up the renderer
+const renderer = new THREE.WebGLRenderer({
+    canvas: document.querySelector("#bg"),
+});
+renderer.setPixelRatio(window.devicePixelRatio);
+renderer.setSize(window.innerWidth, window.innerHeight);
+
+function render() {
+    renderer.render(scene, camera);
+}
+
+const ambientLight = new THREE.AmbientLight();
+scene.add(ambientLight);
+ambientLight.layers.enableAll();
+ambientLight.position.y = 10; //we put the light a bit higher up, otherwise if we put an object at the origin it would block it
+
+const controls = new OrbitControls(camera, renderer.domElement);
+
+function animate() {
+    //this is sort of a main update loop
+    requestAnimationFrame(animate);
+
+    //we also need to update the controls for them to work
+    controls.update();
+
+    render();
+}
+
+animate();
+
+//cache the texture loader
+const textureLoader = new THREE.TextureLoader();
+//cache the GLTF loader
+const gltfLoader = new GLTFLoader();
+
+//we need an array of objects holding all findings for the raycasting to work, otherwise we would be able to select archeological
```

Comment:
Is this objects array holding the objects of interest in an archeological site?

Reply:
Yes, this is for the selection (because we need an array of objects to check for intersections against).